tumblr/kanvas-ios
# Stories: Multiple Export support

- Adds UI for arranging and constructing multiple frames of media.
- Adds support for exporting multiple pieces of media.

## Code

The `MultiEditorViewController` handles displaying of a child `EditorViewController` and `MediaClipsEditorViewController` to display and edit media. It also makes the necessary API calls to start and stop media. There is only ever one `EditorViewController` instance on screen at a time. There are, however, multiple EditorViewControllers at the time of export. Future modifications will refactor the export process to be independent of `EditorViewController` so that this is not necessary.

The `MultiEditorExportHandler` collects the results of the exports from `EditorViewController` and calls its completion block once all have completed.

There is an API change to `didCreateMedia` to pass back multiple media items and errors instead of just one:

```diff
- func didCreateMedia(_ cameraController: CameraController, media: KanvasCameraMedia?, exportAction: KanvasExportAction, error: Error?)
+ func didCreateMedia(_ cameraController: CameraController, media: [(KanvasCameraMedia?, Error?)], exportAction: KanvasExportAction)
```

A new `multipleExport` feature flag is added to enable this feature.

### Testing
* Ensure that the example app works as expected with the feature flag turned off (the default state).
* Enable the `Multiple Export` feature flag and verify that the bottom clips bar shows up after taking a photo:
<a href="https://user-images.githubusercontent.com/3250/98423518-0cd7ba80-204c-11eb-9759-5f5818cbc3cd.jpeg"><img src="https://user-images.githubusercontent.com/3250/98423518-0cd7ba80-204c-11eb-9759-5f5818cbc3cd.jpeg" width="300"></a>
* Tapping the + button on the far right should open the camera screen
* Additional pictures should be added to the media bar at the bottom
* Photos taken with the camera should appear fullscreen _underneath_ the bottom row of pictures:
<a href="https://user-images.githubusercontent.com/3250/98423670-a1421d00-204c-11eb-8eff-507c3abe95aa.jpeg"><img src="https://user-images.githubusercontent.com/3250/98423670-a1421d00-204c-11eb-8eff-507c3abe95aa.jpeg" width="300"></a>
* Tapping the "Post" button should save ALL of the frames to your photos album.
